installer/src/systemd_boot/mod.rs

```diff
@@ -165,9 +165,18 @@ fn remove_old_files(generations: &[Generation], path: &Path) -> Result<()> {
         let f = entry?.path();
         let name = f.file_name().ok_or("filename terminated in ..")?;
 
+        // fwupd puts its own files under "fw" directory, and doesn't remove them
+        // This avoids unhelpful error messages from fs::remove_file later
+        if name == "fw" && f.is_dir() {
```

Comment:
Related to my above comment (but lower in the file... lol): instead of skipping the `fw` directory, maybe we should (offer to) sign the EFI binaries in there instead? I don't think I have an opinion on it being automatic vs. requiring a flag / option to be passed to it.

Reply:
The problem with fwupd is that it expects to have a signed copy of its bootloader in the `/nix/store`; I have yet to do something sensible about it. The best idea I have so far is patching out the check for secureboot from fwupd, and signing it here, but not sure if that wouldn't be too invasive.

Reply:
That's a shame. Thanks for letting me know about that limitation!